test/demo/contextListener.test.ts

```diff
@@ -127,4 +129,53 @@ describe('Context listeners and broadcasting', () => {
             });
         });
     });
+
+    describe('Broadcasting with multiple windows in the same app', () => {
+        const testAppMainWindowIdentity = {uuid: 'test-app-1', name: 'test-app-1'};
+        const testAppChildWindowName = 'test-app-1-child-window';
+        beforeEach(async () => {
+            await fdc3Remote.open(testManagerIdentity, testAppMainWindowIdentity.name);
+        });
+
+        afterEach(async () => {
+            // This `.quit()` closes the main window as well as any child windows
+            await fin.Application.wrapSync(testAppMainWindowIdentity).quit(true);
+        });
+
+        test('When main window broadcasts context, it does not receive its own context, but child window does', async () => {
+            const testAppChildWindowIdentity = await fdc3Remote.createFinWindow(testAppMainWindowIdentity, {url: testAppUrl, name: testAppChildWindowName});
+
+            const testAppMainWindowListener = await fdc3Remote.addContextListener(testAppMainWindowIdentity);
+            const testAppChildWindowListener = await fdc3Remote.addContextListener(testAppChildWindowIdentity);
+
+            await fdc3Remote.broadcast(testAppMainWindowIdentity, validContext);
+
+            // Window broadcasting the context does NOT receive it
+            const testAppMainWindowReceivedContexts = await testAppMainWindowListener.getReceivedContexts();
+            expect(testAppMainWindowReceivedContexts.length).toBe(0);
+
+            // Window on the same app as the one broadcasting DOES receive context
+            const testAppChildWindowReceivedContexts = await testAppChildWindowListener.getReceivedContexts();
+            expect(testAppChildWindowReceivedContexts.length).toBe(1);
+            expect(testAppChildWindowReceivedContexts[0]).toEqual(validContext);
```

Comment:
Would it be simpler to do this as one check - `expect(testAppChildWindowReceivedContexts).toEqual([validContext])`?

Reply:
Very good point, much clearer now. Applied to all tests in the file